engine/runtime-parser/src/main/java/org/enso/compiler/core/IR.java

```diff
@@ -103,6 +104,27 @@ default String pretty() {
   /** Storage for compiler diagnostics related to the IR node. */
   DiagnosticStorage diagnostics();
 
+  /**
+   * Get all diagnostic info associated with this IR node.
+   *
+   * @return the list of diagnostics
+   */
+  List<Diagnostic> diagnosticsList();
+
+  /**
+   * Get the copy of diagnostic storage associated with this IR node.
+   *
+   * @return the copy of diagnostic storage
+   */
+  DiagnosticStorage diagnosticsCopy();
```

Comment:
Why don't we just call `diagnostics().copy()`?

Reply:
I think this makes more sense if you put it into the context of `LazyDiagnosticStorage`, which deal with the possibility of `null`.
Although I would probably already provide a default implementation here:
```
DiagnosticStorage diagnosticCopy() {
  if (diagnostics == null) null else diagnostics.copy();
}
```
Also I would maybe rename it to `diagnosticsLazyCopy`